jetbrains-core/it-202-203/software/aws/toolkits/jetbrains/services/ecr/EcrPushIntegrationTest.kt

```diff
@@ -51,21 +59,56 @@ class EcrPushIntegrationTest {
 
         val project = projectRule.project
         runBlocking {
-            val serverInstance = getDockerServerRuntimeInstance()
+            val serverInstance = getDockerServerRuntimeInstance().runtimeInstance
             val ecrLogin = ecrClient.authorizationToken.authorizationData().first().getDockerLogin()
             val runtime = RemoteDockerApplicationRuntime.createWithPullImage(
                 RemoteDockerRuntime.create(DockerCloudConfiguration.createDefault(), project),
                 DockerAgentBuildImageConfig(System.currentTimeMillis().toString(), dockerfile, false)
             )
             // gross transform because we only have the short SHA right now
             val localImageId = runtime.agent.getImages(null).first { it.imageId.startsWith("sha256:${runtime.agentApplication.imageId}") }.imageId
-            val pushRequest = EcrPushRequest(
+            val pushRequest = ImageEcrPushRequest(
+                dockerServerRuntime = serverInstance,
                 localImageId = localImageId,
                 remoteRepo = remoteRepo,
                 remoteTag = remoteTag
             )
 
-            pushImage(projectRule.project, serverInstance, ecrLogin, pushRequest)
+            pushImage(projectRule.project, ecrLogin, pushRequest)
+        }
+
+        assertThat(
+            ecrClient.describeImages {
+                it.repositoryName(remoteRepo.repositoryName)
+            }.imageDetails()
+        ).anySatisfy {
+            assertThat(it.imageTags()).contains(remoteTag)
+            // would check the digest matches what we've uploaded, but it appears to change upon upload to ECR
```

Comment:
nit: update the comment to mention it's because the image is encrypted/link the doc?

Reply:
updated test to inspect the manifest for the image id